packages/trpc/server/routers/viewer/teams.tsx

```diff
@@ -487,14 +479,21 @@ export const viewerTeamsRouter = createProtectedRouter()
       );
     },
   })
-  .mutation("upgradeTeam", {
+  .mutation("purchaseTeamSubscription", {
     input: z.object({
       teamId: z.number(),
+      billingFrequency: z.union([z.literal("monthly"), z.literal("yearly")]),
+      seats: z.number(),
     }),
     async resolve({ ctx, input }) {
-      if (!HOSTED_CAL_FEATURES)
+      if (!IS_STRIPE_ENABLED)
         throw new TRPCError({ code: "FORBIDDEN", message: "Team billing is not enabled" });
-      return await upgradeTeam(ctx.user.id, input.teamId);
```

Comment:
Why did upgradeTeam cannot be reused?

Reply:
It used a lot of the old logic so I simplified it down in `purchaseTeamSubscription`